src/contracts/protocol/tokenization/RWAAToken.sol

```diff
@@ -0,0 +1,116 @@
+import {SafeCast} from 'src/contracts/dependencies/openzeppelin/contracts/SafeCast.sol';
+import {IERC20} from 'src/contracts/dependencies/openzeppelin/contracts/IERC20.sol';
+import {Errors} from 'src/contracts/protocol/libraries/helpers/Errors.sol';
+import {AToken} from 'src/contracts/protocol/tokenization/AToken.sol';
+import {IRWAAToken} from 'src/contracts/interfaces/IRWAAToken.sol';
+import {IAToken} from 'src/contracts/interfaces/IAToken.sol';
+import {IPool} from 'src/contracts/interfaces/IPool.sol';
+
+abstract contract RWAAToken is AToken, IRWAAToken {
+  using SafeCast for uint256;
+
+  bytes32 public constant ATOKEN_TRANSFER_ROLE = keccak256('ATOKEN_TRANSFER_ROLE');
+
+  /**
+   * @dev Only AToken Transfer Admin can call functions marked by this modifier.
+   */
+  modifier onlyATokenTransferAdmin() {
+    AccessControl aclManager = AccessControl(_addressesProvider.getACLManager());
+    require(
+      aclManager.hasRole(ATOKEN_TRANSFER_ROLE, msg.sender),
+      Errors.CALLER_NOT_ATOKEN_TRANSFER_ADMIN
+    );
+    _;
+  }
+
+  modifier onlyTreasuryRecipient(address recipient) {
+    require(recipient == _treasury, Errors.RECIPIENT_NOT_TREASURY);
+    _;
+  }
+
+  /**
+   * @dev Constructor.
+   * @param pool The address of the Pool contract
+   */
+  constructor(IPool pool) AToken(pool) {
+    // Intentionally left blank
+  }
+
+  /// @inheritdoc AToken
+  function permit(
+    address owner,
+    address spender,
+    uint256 value,
+    uint256 deadline,
+    uint8 v,
+    bytes32 r,
+    bytes32 s
+  ) external virtual override {
+    revert(Errors.OPERATION_NOT_SUPPORTED);
+  }
+
+  /// @inheritdoc IERC20
+  function approve(
+    address spender,
+    uint256 amount
+  ) external virtual override(IERC20, IncentivizedERC20) returns (bool) {
+    revert(Errors.OPERATION_NOT_SUPPORTED);
+  }
+
+  /// @inheritdoc IncentivizedERC20
+  function increaseAllowance(
+    address spender,
+    uint256 addedValue
+  ) external virtual override returns (bool) {
+    revert(Errors.OPERATION_NOT_SUPPORTED);
+  }
+
+  /// @inheritdoc IncentivizedERC20
+  function decreaseAllowance(
+    address spender,
+    uint256 subtractedValue
+  ) external virtual override returns (bool) {
+    revert(Errors.OPERATION_NOT_SUPPORTED);
+  }
+
+  /// @inheritdoc IERC20
+  function transfer(
+    address recipient,
+    uint256 amount
+  ) external virtual override(IERC20, IncentivizedERC20) returns (bool) {
+    revert(Errors.OPERATION_NOT_SUPPORTED);
+  }
+
+  /// @inheritdoc IERC20
+  function transferFrom(
+    address sender,
+    address recipient,
+    uint256 amount
+  ) external virtual override(IERC20, IncentivizedERC20) returns (bool) {
+    revert(Errors.OPERATION_NOT_SUPPORTED);
+  }
+
+  /// @inheritdoc IRWAAToken
+  function forceTransfer(
+    address from,
+    address to,
+    uint256 amount
+  ) external virtual override onlyATokenTransferAdmin returns (bool) {
```

Comment:
Idt it's needed since it's only used once.

Reply:
I think the discussio was about the modifiers -- the fact that we can just inline them, correct?

Reply:
[78c0cfe](https://github.com/aave/aave-v3-horizon/pull/1/commits/78c0cfe181273d953b2a026a5bbbb2dfa152068a)